src/server/scripts/World/go_scripts.cpp

```diff
@@ -1569,6 +1570,123 @@ class go_pirate_day_music : public GameObjectScript
     }
 };
 
+/*####
+## go_bells
+####*/
+
+enum BellHourlySoundFX
+{
+    BELLTOLLHORDE      = 6595, // Horde
+    BELLTOLLTRIBAL     = 6675,
+    BELLTOLLALLIANCE   = 6594, // Alliance
+    BELLTOLLNIGHTELF   = 6674,
+    BELLTOLLDWARFGNOME = 7234,
+    BELLTOLLKHARAZHAN  = 9154 // Kharazhan
+};
+
+enum BellHourlySoundAreas
+{
+    UNDERCITY_AREA     = 1497,
+    IRONFORGE_1_AREA   = 809,
+    IRONFORGE_2_AREA   = 1,
+    DARNASSUS_AREA     = 1657,
+    TELDRASSIL_ZONE    = 141,
+    KHARAZHAN_MAPID    = 532
+};
+
+enum BellHourlyObjects
+{
+    GO_HORDE_BELL      = 175885,
+    GO_ALLIANCE_BELL   = 176573,
+    GO_KHARAZHAN_BELL  = 182064
+};
+
+enum BellHourlyEvent
+{
+    EVENT_RING_BELL    = 1
+};
+
+class go_bells : public GameObjectScript
+{
+public:
+    go_bells() : GameObjectScript("go_bells") { }
+
+    struct go_bellsAI : public GameObjectAI
+    {
+        go_bellsAI(GameObject* go) : GameObjectAI(go), _rings(0) { };
+
+        void OnGameEvent(bool start, uint16 eventId) override
+        {
+            if (eventId == 73 && start == true)
+            {
+                time_t time = sWorld->GetGameTime();
+                tm localTm;
+                localtime_r(&time, &localTm);
+                _rings = (localTm.tm_hour - 1) % 12 + 1;
+                _events.ScheduleEvent(EVENT_RING_BELL, Seconds(1));
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                case EVENT_RING_BELL:
+                {
+                    if (go->GetEntry() == GO_HORDE_BELL)
+                    {
+                        if (go->GetAreaId() == UNDERCITY_AREA)
+                            go->PlayDirectSound(BELLTOLLHORDE);
```

Comment:
maybe precompute this in InitializeAI? a just do go->PlayDirectSound(_soundId);
also go->PlayDirectSound(go->GetAreaId() == UNDERCITY_AREA ? BELLTOLLHORDE : BELLTOLLTRIBAL); one line instead of four